utils/npm/node_modules/three.js/shim.js

```diff
@@ -0,0 +1,12 @@
+var THREE = require('three');
+
+console.warn( "WARNING: The 'three.js' npm package is deprecated in favor of the 'three' npm package, please upgrade.");
+
+if (typeof exports !== 'undefined') {
```

Comment:
The code you linked to exposes three as the return of a require statement, which it would do for line 1 in this file require('three') because that is actually using the 'three.js' file.

But because this shim.js is the main target of the require('three.js') call (e.g. someone using the three.js package), I need to again set the target of that require.  Basically the require('trhee') is nested inside of the require('three.js') and both of these need to set the results of the require statements.

The ```exports``` variable is local to each require and it i returned as the result of the require('xxxx') statement, thus I need to set it up in each require -- doing it within require('three') isn't sufficient for this shim, I need to do it again locally.

Sorry for the long explanation, I think it is a moderately confusing subject.  Full technical details of how require works:

http://wiki.commonjs.org/wiki/Modules/1.1.1

(But there is duplication in the three module now that you point this out -- the code you linked to is duplicated in the footer.js I think.  Harmless, but it is duplication and I should fix that.)